vm/src/compile.rs

```diff
@@ -597,7 +597,20 @@ impl Compiler {
         self.in_loop = false;
         self.in_function_def = true;
         let mut flags = self.enter_function(name, args)?;
-        self.compile_statements(body)?;
+
+        let doc = get_doc(body);
```

Comment:
This code is duplicated with the same code in the class definition part. Try to create a function which performs this split.